File: solutions/csharp/acronym/1/Acronym.cs
using System; 
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        var split = phrase.Split(new[] {' ', '_', '-'}, StringSplitOptions.RemoveEmptyEntries);
        return string.Concat(split.Select(c => Char.ToUpper(c[0])));
    }
} 

File: solutions/fsharp/acronym/2/Acronym.fs
module Acronym

open System

let abbreviate(phrase: string) =
    phrase.Split([| ' '; '-'; '_' |], StringSplitOptions.RemoveEmptyEntries)
    |> Array.map (fun s -> s[0] |> Char.ToUpper |> string)
    |> String.concat ""

